pdns/dnsdist-lua.cc

```diff
@@ -646,6 +648,32 @@ void setupLuaConfig(bool client)
 
     });
 
+  typedef std::unordered_map<std::string, boost::variant<std::string, std::map<std::string, std::string>> > webserveropts_t;
+
+  g_lua.writeFunction("setWebserverConfig", [](boost::optional<webserveropts_t> vars) {
+      setLuaSideEffect();
+
+      if (!vars) {
+        return ;
+      }
+      if(vars->count("password")) {
+        const std::string password = boost::get<std::string>(vars->at("password"));
+
+        setWebserverPassword(password);
+      }
+      if(vars->count("apiKey")) {
+        // allows setting apiKey: nil to disable access with it
```

Comment:
does `nil` really work? It looks like we only test for an empty string?

Reply:
indeed